itaki/rudi
# Keyboard Button Device

There's a new device class in town... 

Introducing **KeyboardButton**!!!

How to test on a Pi:
- Connect an LED on pin 21
- Connect to your Pi (either SSH or directly)
- Pull down the `keyboard-buttons` branch
- Update your local config with the contents of `config-examples/keyboard-led.json`
- Set `use_mock_pins` to false in your config
- From `rudi-app` folder, run: `docker compose build --no-cache && docker compose run rudi-app`
- See that the `l` key causes the light to turn on
- See that the `p` key causes the light to turn off

Other notes:
- This also works outside of docker, just be sure to install dependencies from requirements.txt
- Only simple/normal keys will work (things like shift key and spacebar key won't work)
